internal/dag/gatewayapi_processor.go

```diff
@@ -1060,10 +1060,10 @@ func (p *GatewayAPIProcessor) computeHTTPRoute(route *gatewayapi_v1beta1.HTTPRou
 
 		// Process rule-level filters.
 		var (
-			headerPolicy       *HeadersPolicy
-			headerModifierSeen bool
-			redirect           *gatewayapi_v1beta1.HTTPRequestRedirectFilter
-			mirrorPolicy       *MirrorPolicy
+			requestHeaderPolicy, responseHeaderPolicy             *HeadersPolicy
+			requestHeaderModifierSeen, responseHeaderModifierSeen bool
```

Comment:
not directly related to your PR but seems like we could drop the bools and just nil-check the policy vars